src/main/java/IDriver.java

```diff
@@ -0,0 +1,45 @@
+public interface IDriver {
+    /**
+     * to call the FSDB to create a new database
+     * @param schemaPath the schema of the database required to be created
+     */
+    void createDatabase (String schemaPath);
+
+    /**
+     * to set a new row (element) in the database
+     * @param databaseName the name of database to set in the new row
+     * @param jsonObject the row wanted to be set in the database
+     */
+    void set (String databaseName, String jsonObject);
+
+    /**
+     * to get a row form the database
+     * @param databaseName the name of database to get form it the row
+     * @param jsonQuery the query data of required row
+     * @return list of json of row that match the query
+     */
+    String get (String databaseName, String jsonQuery);
+
+    /**
+     *
+     * @param databaseName the name of database to delete form it the row
+     * @param jsonQuery the query data of required row
+     */
+    void delete (String databaseName, String jsonQuery);
+
+    // here we will get the output from the python then pass it to java output function
+    // it will take it and check the status
+    //      if any exception will call exceptionRaiser and give it the message and exception name
+    //      or success and get function return the result
+    //      print the message
+    //private method
+    void output (String jsonResult);
```

Comment:
- Why does this method (output) exist in the interface?
 The interface for the public method only.

same as exception raiser.

why do we need result ?

Reply:
have no idea than interface for public methods only will remove them.

Reply:
need the result because it is the function that will pass the result of the get to the user it may be implemented for now by just console print but in the future, I may change the implementation to a web page or GUI so it takes the json list of result and prints it to the user as it defined (GUI, console,..)

Reply:
and it is private too so i should remove it from here